### ISSUE TITLE
Accessibility issues on the search results page

### ISSUE THREAD

**Issue by selfthinker** [splitbrain/dokuwiki#3460]
Apart from #3459, there is only one more fail on the [search results page](https://grimm.haz.wiki/start?do=search&id=start&q=king) and a couple more non-fail issues.

* [ ] The main search field doesn't have an accessible name. It needs to have a `label`, ideally visible (but it can also be visually hidden or have an `aria-label` etc).
That fails [4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value) and has a medium priority.

![](https://user-images.githubusercontent.com/108893/113491212-f3d21900-94c6-11eb-8cc2-6f409763b87c.png)

## Other non-fail accessibility issues

These don't strictly fail WCAG but are still accessibility issues. Fixing them is recommended but optional for passing WCAG.
These are all minor except for the magnification issue.

* [ ] The `fieldset` around the search field and filters doesn't have a `legend`.

* [ ] It's confusing and less discoverable seeing the search form when magnifying the screen because everything is centred. (Similar to what I reported about the login form in #3457.) The fieldset border and the input border look too similar, and when you cannot see the end of either, it's difficult to understand what it is. That's made worse because the input is missing a label and the fieldset is missing a legend, which makes them look less like form elements.
I would add the label and legend and left-align the form.
![](https://user-images.githubusercontent.com/108893/113491396-65f72d80-94c8-11eb-9ddd-27a2a15fcd1e.png)

* [ ] The search terms are highlighted via the `strong` element. That is okay, but using the `mark` element would be more semantic (new since HTML5). Screen readers never announce `strong` but some announce `mark`.

* [ ] The search terms are not highlighted when in High Contrast Mode or when changing colours in Firefox. Using `mark` instead of `strong` (see above) would fix that. But this could also be fixed through other means, for example by giving the highlighting an invisible border or outline.
![](https://user-images.githubusercontent.com/108893/113491580-de122300-94c9-11eb-9661-6f350d6e31b0.png)
![](https://user-images.githubusercontent.com/108893/113491581-deaab980-94c9-11eb-81d8-9ca89115d406.png)

* [ ] The 'toggle search tools' button comes visually after the search field and button but comes before them in source and tab order. That's a bit confusing and unexpected.

* [ ] There are two 'search' buttons on the page, one in the header (the magnifier icon) and the other in the main content's search. Although they perform the same function, they are on two different forms and are not interchangeable. It would be better (for voice control and screen reader users) to give them unique names. Although I'm not sure with which words to differentiate them and if that might cause more confusion.

* [ ] The search results are using a description list (`dl` with `dt` and `dd`). Although that is semantically fine, unfortunately screen readers don't handle them very well and they are read out quite inconsistently. For example, NVDA just reads lines, and VoiceOver repeats "term" and especially "description" a lot. That's more something to be aware of and not necessarily to change.

* [ ] It is unclear what the purpose of the `@stories` links are and they make even less sense out of context. But they are equally unclear to everyone, which makes this a usability rather than an accessibility issue.
